File: 647-palindromic-substrings/palindromic-substrings.cpp
class Solution {
public:

bool isPal(const string &s, int l, int r) {
    while (l < r) {
        if (s[l] != s[r]) return false;
        l++;
        r--;
    }
    return true;
}

int countSubstrings(string s) {
    int count = 0;
    int n = s.size();
    
    for (int i = 0; i < n; i++) {
        for (int j = i; j < n; j++) {   
            if (isPal(s, i, j)) count++;
        }
    }
    return count;
}



};

File: 647-palindromic-substrings/palindromic-substrings.java
class Solution {
    static boolean isPal(String s, int l, int r){
        while(l<r){
            if(s.charAt(l)!=s.charAt(r))return false;
            l++;
            r--;
        }
        return true;
    }
    public static int countSubstrings(String s) {
        int n = s.length();
        int count = 0;
        for(int i=0; i<n; i++){
            for(int j = n-1; j>=i; j--){
                if(isPal(s,i,j))count++;
            }
        }
        return count;
    }
}